### ISSUE TITLE
Improve `Style/RedundantParentheses` cop for lambdas

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12380]
## Actual behavior

I have the following code:

```ruby
class Post < ApplicationRecord
  scope :published1, (lambda do
    foo
  end)

  scope :published2, (-> { bar })
end
```

Terminal:
```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/post.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Rubocop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.57.2 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by ydakuka** [comment]
And for the next examples there is a wrong indentation, but rubocop does not detect any offences.

```ruby
class User < ApplicationRecord
  scope :published3, (lambda do
                        baz
                      end)

  scope :published4, (
        -> { car }
      )
end
```
